### PR TITLE
[PKG-5556] snowflake 0.12.0 ❄️ 

### PR DESCRIPTION
**Pull request by psteyer**
snowflake 0.12.0  :snowflake:

**Destination channel:** Snowflake 

### Links

- [PKG-5556](https://anaconda.atlassian.net/browse/PKG-5556) 
- dependency PR's:
    - https://github.com/AnacondaRecipes/snowflake_legacy-feedstock/pull/4
    - https://github.com/AnacondaRecipes/snowflake.core-feedstock/pull/6 

### Explanation of changes:

- build number update
- update to 0.12.0

[PKG-5556]: https://anaconda.atlassian.net/browse/PKG-5556?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lorepirri** [comment]
@psteyer could you please set CODEOWNERS to `@AnacondaRecipes/psubs` in `.github/CODEOWNERS`?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: e85d9bc72fd4cac2d764bfb57032a1a9b51d514e34ddede74aeda7635f7fa5b4
+  sha256: 9a8c0fb1ac1a2720a0c913c3c1cd9151267f9d3c4373d2d2055a93281ec5ca28
   patches:
     # update 0.10.0: patch still needed
     - patches/0001-fix-pyproject.patch
```
Comment (lorepirri) [inline recipe/meta.yaml:13]: is this patch still needed for this version?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 
 build:
-  number: 0
+  number: 100
```
Comment (lorepirri) [inline recipe/meta.yaml:16]: could you please add a comment about why we set this to 100?

**Review by lorepirri — APPROVED** [review]

**Comment by ifitchet** [comment]
the `pyproject.toml` sets an optional dependency https://inspector.pypi.io/project/snowflake/0.12.0/packages/75/6c/a3c3e7474d62350a3ca302a91359529098d23b10e49c5fdde9fa1852a0a2/snowflake-0.12.0.tar.gz/snowflake-0.12.0/pyproject.toml#line.49

Do we _not_ set that as a `run_constrained`?

File: recipe/meta.yaml (hunk 2 above)
Comment (psteyer) [reply recipe/meta.yaml:18]: added comment

File: recipe/meta.yaml (hunk 1 above)
Comment (psteyer) [reply recipe/meta.yaml:13]: patch seems to be still required. I updated the comment.

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (lorepirri) [reply recipe/meta.yaml:13]: thanks!

**Pull request merged**